executable/java_class.ksy

```diff
@@ -83,28 +88,85 @@ types:
             'tag_enum::method_handle': method_handle_cp_info
             'tag_enum::method_type': method_type_cp_info
             'tag_enum::invoke_dynamic': invoke_dynamic_cp_info
+            'tag_enum::dynamic': dynamic_cp_info
+            'tag_enum::module': module_package_cp_info
+            'tag_enum::package': module_package_cp_info
         if: not is_prev_two_entries
     instances:
       is_two_entries:
         value: 'is_prev_two_entries ? false : tag == tag_enum::long or tag == tag_enum::double'
     enums:
       tag_enum:
-        7: class_type
-        9: field_ref
-        10: method_ref
-        11: interface_method_ref
-        8: string
-        3: integer
-        4: float
-        5: long
-        6: double
-        12: name_and_type
-        1: utf8
-        15: method_handle
-        16: method_type
-        18: invoke_dynamic
+        1:
+          id: utf8
+          -orig-id: CONSTANT_Utf8
+        3:
+          id: integer
+          -orig-id: CONSTANT_Integer
+        4:
+          id: float
+          -orig-id: CONSTANT_Float
+        5:
+          id: long
+          -orig-id: CONSTANT_Long
+        6:
+          id: double
+          -orig-id: CONSTANT_Double
+        7:
+          id: class_type
+          -orig-id: CONSTANT_Class
+        8:
+          id: string
+          -orig-id: CONSTANT_String
+        9:
+          id: field_ref
+          -orig-id: CONSTANT_Fieldref
+        10:
+          id: method_ref
+          -orig-id: CONSTANT_Methodref
+        11:
+          id: interface_method_ref
+          -orig-id: CONSTANT_InterfaceMethodref
+        12:
+          id: name_and_type
+          -orig-id: CONSTANT_NameAndType
+        15:
+          id: method_handle
+          -orig-id: CONSTANT_MethodHandle
+        16:
+          id: method_type
+          -orig-id: CONSTANT_MethodType
+        17:
+          id: dynamic
+          -orig-id: CONSTANT_Dynamic
+        18:
+          id: invoke_dynamic
+          -orig-id: CONSTANT_InvokeDynamic
+        19:
+          id: module
+          -orig-id: CONSTANT_Module
+        20:
+          id: package
+          -orig-id: CONSTANT_Package
+
+  version_guard:
+    doc: |
+      `class` file format version 45.3 (appeared in the very first publicly known release of Java SE 1.0.2 and JDK 1.0, released 23th January 1996) is so ancient that it's taken for granted. Earlier formats seem to be undocumented. Changes of `version_minor` don't change `class` format.
+      Earlier `version_major`s likely belong to Oak programming language, the proprietary predecessor of Java.
+    doc-ref:
+      - James Gosling, Bill Joy and Guy Steele. The Java Language Specification. English. Ed. by Lisa Friendly. Addison-Wesley, Aug. 1996, p. 825. ISBN: 0-201-63451-1.
+      - Frank Yellin and Tim Lindholm. The Java Virtual Machine Specification. English. Ed. by Lisa Friendly. Addison-Wesley, Sept. 1996, p. 475. ISBN: 0-201-63452-X.
```

Comment:
This is a key-value pair in YAML. I hope it wouldn't slip through the compiler :grimacing:

```suggestion
    doc-ref:
      - 'James Gosling, Bill Joy and Guy Steele. The Java Language Specification. English. Ed. by Lisa Friendly. Addison-Wesley, Aug. 1996, p. 825. ISBN: 0-201-63451-1.'
      - 'Frank Yellin and Tim Lindholm. The Java Virtual Machine Specification. English. Ed. by Lisa Friendly. Addison-Wesley, Sept. 1996, p. 475. ISBN: 0-201-63452-X.'
```

Reply:
> I hope it wouldn't slip through the compiler 😬

Fortunately, it doesn't:

```console
$ kaitai-struct-compiler-0.10 -- --verbose file -t java java_class.ksy
parsing java_class.ksy...
reading java_class.ksy...
java_class.ksy: /types/version_guard/doc-ref/0:
        error: expected string, got Map(James Gosling, Bill Joy and Guy Steele. The Java Language Specification. English. Ed. by Lisa Friendly. Addison-Wesley, Aug. 1996, p. 825. ISBN -> 0-201-63451-1.) (class scala.collection.immutable.MapLike$$anon$2)
```